packages/flutter/test/services/binding_test.dart

```diff
@@ -0,0 +1,66 @@
+import 'dart:convert';
+import 'dart:typed_data';
+
+import 'package:flutter/foundation.dart';
+import 'package:flutter/scheduler.dart';
+import 'package:flutter/services.dart';
+import 'package:flutter/src/services/binding.dart';
+
+import 'package:flutter_test/flutter_test.dart';
+
+const String license1 = '''
+L1Package1
+L1Package2
+L1Package3
+
+L1Paragraph1
+
+L1Paragraph2
+
+L1Paragraph3''';
+
+const String license2 = '''
+L2Package1
+L2Package2
+L2Package3
+
+L2Paragraph1
+
+L2Paragraph2
+
+L2Paragraph3''';
+
+const String licenses = '''
+$license1
+--------------------------------------------------------------------------------
+$license2
+''';
+
+class TestBinding extends BindingBase with SchedulerBinding, ServicesBinding {
+  @override
+  BinaryMessenger createBinaryMessenger() {
+    return super.createBinaryMessenger()
+      ..setMockMessageHandler('flutter/assets', (ByteData message) async {
+        if (utf8.decode(message.buffer.asUint8List()) == 'LICENSE') {
+          return ByteData.view(Uint8List.fromList(utf8.encode(licenses)).buffer);
```

Comment:
This could probably be simplified with `StandardMessageCodec().encodeMessage(...)`.

Reply:
I don't think so because the asset bundle does not use the standard message codec decode method. The `StandardMessageCodec` adds a type discriminator byte first but the `AssetBundle.loadString` just tries to utf8 decode the bytes without striping the discriminator byte.

Reply:
Just found out that there is a StringCodec I can use

Reply:
The AssetBundle should probably make use of this message codec too. I will open a separate PR for this!